model/log_iterator.go

```diff
@@ -29,6 +29,8 @@ type LogIterator interface {
 	Err() error
 	// Item returns the current LogLine item held by the iterator.
 	Item() LogLine
+	// Reverse reverses the iterator.
+	Reverse() error
```

Comment:
I feel like this should return a LogIterator or you should be able to pass an option which is either "FORWARD" or "REVERSE" and have it error if you've started iterating?